dds/idl/be_produce.cpp

```diff
@@ -289,11 +289,22 @@ BE_produce()
   ifstream idl(idl_fn);
   const size_t buffer_sz = 512;
   char buffer[buffer_sz];
-
+  unsigned lineno = 0;
+  bool warned_dcps_data_type =
+    be_global->no_dcps_data_type_warnings_ ||
+    !idl_global->print_warnings();
   while (idl) {
     idl.getline(buffer, buffer_sz);
-
-    if (0 == strncmp("#include", buffer, 8)) { //FUTURE: account for comments?
+    lineno++;
+
+    if (!(warned_dcps_data_type || strncmp("#pragma DCPS_DATA_TYPE", buffer, 22))) {
```

Comment:
Yes, absolutely this could be improved, I just didn't think they would have to go into this release. I have some ideas about having tao_idl help with this that are going to be part of the annotation migrator. If you want me to make this more flexible for this release I can.